client/optimizing.go

```diff
@@ -324,6 +324,7 @@ func parallelGet(ctx context.Context, clients []Client, round uint64, timeout ti
 		wg := sync.WaitGroup{}
 	LOOP:
 		for _, c := range clients {
+			c := c
```

Comment:
The `c` type here is a `Client interface` value. So, we don't know what's implementing the `Client` type, it's better to pass this by value rather than reference.